src/app/test-record/test-record.mapper.ts

```diff
@@ -212,7 +112,8 @@ export class TestRecordMapper {
     testTypeMapped.emissionStandard = testResultFormData.testType.emissionStandard;
     testTypeMapped.smokeTestKLimitApplied = testResultFormData.testType.smokeTestKLimitApplied;
     testTypeMapped.fuelType = testResultFormData.testType.fuelType;
-    if (!!testTypeMapped.modType) {
+
+    if (!!testTypeMapped.modType && testResultMapped.vehicleType === 'trl') {
```

Comment:
Let's please log a task to cover that later as the mapper unit tests seem to be covering not all the permutations/logic we have.
We will cover that later if we have to ship it now please.

Reply:
https://jira.dvsacloud.uk/browse/CVSB-16228